src/pybind/mgr/progress/module.py

```diff
@@ -542,6 +581,10 @@ def update(self, ev_id, ev_msg, ev_progress, refs=None):
         ev.set_progress(ev_progress)
         ev._refresh()
 
+    def add_event(self, ev):
```

Comment:
do we have any caller of this method?

Reply:
So in the balancer module, when the upmap optimization gets triggered, it will need to add the event to the _event dict in progress module. I did this to avoid having duplicate _event variables.